build.rs

```diff
@@ -33,17 +33,22 @@ fn main() {
                 None => "clang-16".into(),
             };
 
+            build.compiler(clang);
+        }
+
+        let compiler = build.get_compiler();
+        if compiler.is_like_clang() {
             build
-                .compiler(clang)
```

Comment:
It's not changed in this PR but `no_default_flags` means that cc won't automatically add flags about optimization or debug info.

Reply:
We probably want to set optimization and debug flags according to the OPT_LEVEL and DEBUG variables same as before.

Reply:
Updated, nice catch.